rancher_pr_deploy.sh

```diff
@@ -13,14 +13,31 @@ set -o pipefail
 
 declare -r NHSUK_GITHUB_URL="https://api.github.com/repos/nhsuk"
 
+find_latest_rancher_template() {
+
+  TEMPLATE_URL_BASE=$1
+  TEMPLATE_VERSION=0
```

Comment:
We already have a function with checks gets the response for a URL so the function could be simplified to:

```
find_latest_rancher_template() {

  declare -r TEMPLATE_URL_BASE=$1
  declare TEMPLATE_VERSION; TEMPLATE_VERSION=0

  until [ "$(get_http_response "${TEMPLATE_URL_BASE}/${TEMPLATE_VERSION}/docker-compose.yml")" != "200" ]; do
    TEMPLATE_VERSION=$((TEMPLATE_VERSION + 1))
  done

  echo $((TEMPLATE_VERSION - 1))
}
```